src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -57,15 +57,23 @@ class LedgerImpl(vm: VM, blockchainConfig: BlockchainConfig) extends Ledger with
     val blockExecResult = for {
       //todo why do we require in executeBlockTransactions separate blockchain and blockchain storage
       //todo blockchain storage is part of blockchain
-      execResult <- executeBlockTransactions(block, blockchain, storages, validators.signedTransactionValidator).left.map(e => TxError(e.reason))
+      execResult <- executeBlockTransactions(block, blockchain, storages, validators.signedTransactionValidator)
       BlockResult(resultingWorldStateProxy, _, _) = execResult
       worldToPersist = payBlockReward(blockchainConfig.blockReward, block, resultingWorldStateProxy)
       worldPersisted = InMemoryWorldStateProxy.persistState(worldToPersist) //State root hash needs to be up-to-date for prepared block
-    } yield BlockPreparationResult(execResult, worldPersisted.stateRootHash)
-
-    blockExecResult.foreach { _ => log.debug(s"Prepared for mining block with number ${block.header.number}") }
-
-    blockExecResult
+    } yield BlockPreparationResult(block, execResult, worldPersisted.stateRootHash)
+
+    blockExecResult match {
+      case Right(result) =>
+        log.debug(s"Prepared for mining block with number ${block.header.number}")
+        Right(result)
+      case Left(TxsExecutionError(tx,reason)) =>
+        log.debug(s"failed to execute transaction $tx because: $reason")
+        val newBlock = block.copy(body = block.body.copy(transactionList = block.body.transactionList.filter(_.hash != tx.hash)))
```

Comment:
this is good question, it may be that after future block reorganisation it will not fail
and we have limited transaction pool so failing transactions will be push out by new one